fahadmak/django_iot_dashboard
# #167556577 Set up user authentication

**What does this PR do?**
This PR ensures that a user can log in, log out and sign up into the application.

**Description of Task to be completed?**
- create User models
- create login and sign up templates
- create User test_forms
- create login, sign up and logout views

**What are the relevant Pivotal Tracker stories?**
[#167556577](https://www.pivotaltracker.com/n/projects/2364284/stories/167556577)

**How should be tested manually?**
Run each of the following commands in your terminal in the respective order.

- `git clone https://github.com/fahadmak/stackoverflow_clone.git`
- `cd django_iot_dashboard`
- `git checkout ft-user-authentication-167556577`
- `virtualenv venv`
- `. venv/bin/activate`
- `pip install -r requirements.txt`
- `run migrations using this command python manage.py migrate`
- `run python manage.py runserver 127.0.0.1:8000`
- `Go to this link in your browser 127.0.0.1:8000`
- `You will be redirected to the login page where you should log in`
- If you have no account, click the sign up link to go sign up page`

**Screenshots**
***Login***
<img width="1021" alt="Room Availability - 600dp After" src="https://user-images.githubusercontent.com/13882936/62106489-00dc4500-b2ae-11e9-93ff-9af2a7ac70e8.png">

***Sign up***
<img width="1021" alt="Room Availability - 600dp After" src="https://user-images.githubusercontent.com/13882936/62106491-0174db80-b2ae-11e9-8863-1d62947673eb.png">
